src/main/java/org/bytedeco/javacv/FFmpegFrameGrabber.java

```diff
@@ -1325,6 +1329,27 @@ public Frame grabFrame(boolean doAudio, boolean doVideo, boolean doProcessing, b
                         frame.keyFrame = samples_frame.key_frame() != 0;
                     }
                 }
+            } else {
+                // Create and grow the streams array as necessary to avoid using HashMap 
+                if(other_idx.length < (pkt.stream_index()+1)) {
+                    int[] new_idx = new int[pkt.stream_index()+1];
+                    Arrays.fill(new_idx, -1);
+                    System.arraycopy(other_idx, 0, new_idx, 0, other_idx.length);
+                    other_idx = new_idx;
+                } 
+                if(other_idx[pkt.stream_index()] == -1) {
+                    other_idx[pkt.stream_index()] = num_other_streams;
+                    num_other_streams++;
+                }
+                if((frame.otherStreams == null) || (frame.otherStreams.length < num_other_streams)) {
+                    ByteBuffer[] other_streams = new ByteBuffer[num_other_streams];
+                    if(frame.otherStreams != null) {
+                       System.arraycopy(frame.otherStreams, 0, other_streams, 0, frame.otherStreams.length);
+                    }
+                    frame.otherStreams = other_streams;
+                }
+                ByteBuffer bb = pkt.data().position(0).limit(pkt.size()).asByteBuffer();
+                frame.otherStreams[num_other_streams-1] = bb;
```

Comment:
I don't understand the logic. How do the buffers for all other data streams than the last one get filled?

Reply:
You're right; that incorrect indexing came in with some mis-merging on my end.  That line should be `frame.otherStreams[other_idx[pkt.stream_index()]] = bb;`

Reply:
Ok, that's better. So, if I understand correctly, each data packet doesn't have more than 1 buffer anyway, so we don't actually need to use an array for `Frame.data` either, right? Having just a `Buffer data` field and putting the data for that packet/frame there would get us the same thing, correct?

Reply:
From further reading of the code it appears my mental model of how this works was not quite right.  So yes, having an array for Frame.data doesn't quite make sense (or even work correctly without cloning data before storage).  I will rework this portion to export a single ByteBuffer and stream index.